src/query/service/src/interpreters/interpreter_privilege_grant.rs

```diff
@@ -185,9 +205,34 @@ impl Interpreter for GrantPrivilegeInterpreter {
                     .await?;
             }
             PrincipalIdentity::Role(role) => {
+                if role == BUILTIN_ROLE_ACCOUNT_ADMIN || role == BUILTIN_ROLE_PUBLIC {
+                    return Err(databend_common_exception::ErrorCode::PermissionDenied(
+                        "cannot grant privileges to buildin roles",
```

Comment:
I guess it's allowed to grant privileges to PUBLIC, when we need exposes some privilege to everyone.

Reply:
Yes. This is a mistake. only grant ownership to built-in roles needs a forbidden.

Reply:
IMHO public & account_admin could also be granted as owner

Reply:
Done